data/src/main/java/org/springframework/security/data/repository/query/SecurityEvaluationContextExtension.java

```diff
@@ -77,12 +84,21 @@
  * it.
  *
  * @author Rob Winch
+ * @author Evgeniy Cheban
  * @since 4.0
  */
 public class SecurityEvaluationContextExtension implements EvaluationContextExtension {
 
 	private Authentication authentication;
 
+	private AuthenticationTrustResolver trustResolver = new AuthenticationTrustResolverImpl();
+
+	private RoleHierarchy roleHierarchy = new NullRoleHierarchy();
```

Comment:
This seems fine, but I think the `RoleHierarchy` can be `null` by default, since it is checked internally in the `SecurityExpressionRoot` and only used if provided. What do you think?

Reply:
Agreed. Perhaps, we should also consider adding a `null` check for `trustResolver` in `SecurityExpressionRoot` or assign a default value.

Reply:
Thanks for working on this @evgeniycheban. Sadly, I wasn’t able to follow up on this quickly enough. I did have a conversation with the team about this and learned that Spring Security conventions prefer defaults over null checks. I’m still newer on the team so I’m still learning! So what you had in both of these cases was preferred. Would you mind force pushing 2bfd933 back (unless you applied any polish). I’m sorry for the trouble.

Reply:
Done.